### PR TITLE
fixed JSON schema id 

### PR DESCRIPTION
**Pull request by jkowalleck**
is part  to #83 

* removed duplicate `$id` definitions
* removed all `$id` markers that were just json-pointers prefixed with `#` (self-shadowing)

### CODE REVIEW

**Comment by jkowalleck** [comment]
hm, CI is breaking.
any idea why?

**Comment by stevespringett** [comment]
The build was relying on an old SNAPSHOT of CycloneDX Core Java which was used at the time of the release of the v1.3 spec. The build needed to be updated to a released version of Core Java, which I have just done. Future builds should pass.

**Comment by jkowalleck** [comment]
thanks, @stevespringett .
rebased onto latest master; build passed as expected.

**Comment by jkowalleck** [comment]
@stevespringett could you revisit this PR?

**Comment by stevespringett** [comment]
@jkowalleck We're currently working on the v1.4-dev branch and already have 1 PR submitted. I'll likely have another one this weekend. But I'd like to include all of these fixes in the v1.4-dev branch when everything is merged into that branch prior to merging to master.

**Comment by jkowalleck** [comment]
need to give tis one a review. wait a minute

File: schema/bom-1.3-strict.schema.json [hunk 1]
```diff
@@ -834,7 +820,6 @@
           ]
         },
         "hashes": {
-          "$id": "#/properties/hashes",
```
Comment (jkowalleck) [inline schema/bom-1.3-strict.schema.json:837]: was a duplicate.

**Comment by stevespringett** [comment]
Fixed in 1.4. Core Working Group has not made a decision on whether to backport.

**Comment by jkowalleck** [comment]
related to #123

File: schema/bom-1.2-strict.schema.json [hunk 2]
```diff
@@ -27,14 +26,12 @@
       ]
     },
     "specVersion": {
-      "$id": "#/properties/specVersion",
```
Comment (jkowalleck) [inline schema/bom-1.2-strict.schema.json:30]: `$id` is the same value the JSON pointer has. so the `$id` is shadowing the exact same element and can therefore be removed.

**Comment by stevespringett** [comment]
Closing in favor of #125

**Pull request closed without merging**